src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/RethrowToPreserveStackDetailsTests.cs

```diff
@@ -495,6 +496,68 @@ End Class
 ");
         }
 
+        [Fact]
+        public async Task CA2200_DiagnosticsForThrowCaughtExceptionInLocalMethod()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+
+class Program
+{
+    void CatchAndRethrowExplicitly()
+    {
+        try
+        {
+        }
+        catch (Exception e)
+        {
+
+            DoThrow();
+
+            void DoThrow()
+            {
+                [|throw e;|]
```

Comment:
This indicates there is a bug. Likely following will show a violation, when it should not:
```csharp
    void CatchAndRethrowExplicitly()
    {
        try
        {
        }
        catch (Exception e)
        {
            e = new MyException();
            DoThrow();
            void DoThrow()
            {
                throw e; // This should not fire.
            }
        }
    }
```